### ISSUE TITLE
when this will be upstream to mainline

### ISSUE THREAD

**Issue by userforsource** [linux-surface/linux-surface#1419]
Having this in mainline will support in ubuntu fedira opensuse as they provide signed kernel so we can have a secure boot or verified boot support which is necessary for now a days.

**Comment by qzed** [comment]
We are working on it. See https://github.com/linux-surface/linux-surface/issues/205 for details.

I think most functionality should be usable upstream on most devices, except for touch.

**Comment by userforsource** [comment]
Yes i think this stuff should be asap like see apple m1 m2 m3 stuff has been upstreamed now you can use fedora asahi remix. And so on...

**Comment by qzed** [comment]
We're generally pushing it upstream as soon as we think it's in shape for that. Unlike Asahi, however, we don't have people working full-time on this. Both @StollD and I do this in our free time (of which I haven't had all that much lately), meaning it might take some time to get there. But we are working on it.

**Comment by userforsource** [comment]
I highly appreciate that i know that.
Upstream this actually helps many this i hope you understand thanks i hope you will help upstream asap. Stuffs gets stable. and this is mostly for the newer devices.
